build/Makefile

```diff
@@ -126,6 +126,7 @@ clean : clean-status
 clean-ruby : clean
 	sudo $(RMDIR) $(BASE_DIR)/intermediate
 	sudo $(RMDIR) $(RUBY_DIR)/.ext
+	sudo $(RMDIR) /usr/local/ruby-2.2.0*
```

Comment:
The problem with this sort of approach is that we don't always do a 'make clean' to clean things up. We almost exclusively just use git to clean things up (i.e. `git clean` with options, which is what `git nuke` does).

If this is your intent, I think a much better way is to simply eliminate the copy of Ruby in /usr/local entirely, instead building Ruby in the users local directory (i.e. intermediate/*/local_ruby). This way, `git nuke` would clean it up as well.

I still believe that the better way is to save a checksum, and if something key changes, then rebuild ruby. This would get rid of the notion of a letter in the directory; instead, we'd recognize that it would need to be rebuilt and just rebuild it.

But if you don't want to do that, then building Ruby on a per-user basis in the intermediate directory is a better way to go. Then `git nuke` would clean it up, which is what the build system generally does anyway.